streaming/base/storage/download.py

```diff
@@ -158,19 +164,19 @@ def download_from_gcs(remote: str, local: str) -> None:
         remote (str): Remote path (GCS).
         local (str): Local path (local filesystem).
     """
+    from google.auth.exceptions import DefaultCredentialsError
     obj = urllib.parse.urlparse(remote)
     if obj.scheme != 'gs':
         raise ValueError(
             f'Expected obj.scheme to be `gs`, instead, got {obj.scheme} for remote={remote}')
 
-    if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
-        _gcs_with_service_account(local, obj)
-    elif 'GCS_KEY' in os.environ and 'GCS_SECRET' in os.environ:
+    if 'GCS_KEY' in os.environ and 'GCS_SECRET' in os.environ:
         _gcs_with_hmac(remote, local, obj)
     else:
-        raise ValueError(f'Either GOOGLE_APPLICATION_CREDENTIALS needs to be set for ' +
-                         f'service level accounts or GCS_KEY and GCS_SECRET needs to be ' +
-                         f'set for HMAC authentication')
+        try:
+            _gcs_with_service_account(local, obj)
+        except (DefaultCredentialsError, EnvironmentError):
+            raise ValueError(GCS_ERROR_NO_AUTHENTICATION)
```

Comment:
Please add a test for this error

Reply:
Added the test in a6cf8d2fac3b684554039526aa46f6b2dac1d4bb